html/modules/custom/reliefweb_import/src/Command/ReliefwebImportCommand.php

```diff
@@ -0,0 +1,680 @@
+    }
+  }
+
+  /**
+   * Validate title.
+   */
+  protected function validateTitle($title) {
+    if (empty(trim($title))) {
+      throw new ReliefwebImportExceptionViolation('Job found with empty title.');
+    }
+
+    // Clean the title.
+    $title = strip_tags($title);
+
+    $options = [
+      'line_breaks' => TRUE,
+      'consecutive' => TRUE,
+    ];
+    $title = $this->cleanText($title, $options);
+
+    // Ensure the title size is reasonable. The max length matches the one from
+    // the job form.
+    $length = mb_strlen($title);
+    if ($length < 10 || $length > 150) {
+      throw new ReliefwebImportExceptionViolation('Invalid title length.');
+    }
+
+    return $title;
+  }
+
+  /**
+   * Validate the body field of a feed item, clean and check its size.
+   *
+   * @param string $data
+   *   Raw data from XML.
+   */
+  protected function validateBody($data) {
+    // Clean the body field.
+    $body = $this->sanitizeText('body', $data, 'plain_text');
+
+    // Ensure the body field size is reasonable.
+    $length = mb_strlen($body);
+    if ($length < 400 || $length > 50000) {
+      throw new ReliefwebImportException(strtr('Invalid field size for body, @length characters found, has to be between 400 and 50000', [
+        '@length' => $length,
+      ]));
+    }
+
+    return $body;
+  }
+
+  /**
+   * Sanitize text, converting it to markdown.
+   *
+   * @param string $field
+   *   Field name.
+   * @param string $text
+   *   Field text content.
+   * @param string $format
+   *   Format to which conver the text if not already HTML.
+   *
+   * @return string
+   *   Sanitized content.
+   */
+  protected function sanitizeText($field, $text, $format = 'plain_text') {
+    if (!is_string($text)) {
+      return '';
+    }
+
+    // Trim the input text.
+    $text = trim($text);
+    if (empty($text)) {
+      return '';
+    }
+
+    // Sometimes the parser bugs and leaves the closing field tag at the end
+    // of the text. We remove it before proceeding.
+    $position = mb_strpos($text, '</' . $field . '>');
+    if ($position !== FALSE) {
+      $text = mb_substr($text, 0, $position);
+    }
+
+    // Clean the text, removing notably control characters.
+    $text = $this->cleanText($text);
+
+    // Check if the text is wrapped in a CDATA.
+    if (mb_stripos($text, '<![CDATA[') === 0) {
+      $end = mb_strpos($text, ']]>');
+      $text = mb_substr($text, 9, $end !== FALSE ? $end - 9 : NULL);
+    }
+
+    // Check if the content contains some non encoded html tags, in which case
+    // we will assume that the text is non encoded html/markdown. For that we
+    // simply search for a closing tag '</...>'. Otherwise we decode the text.
+    if (preg_match('#</[^>]+>#', $text) !== 1) {
+      $text = html_entity_decode($text, ENT_QUOTES, 'UTF-8');
+    }
+
+    // Convert the text to HTML.
+    /*
+    if (function_exists('check_markup')) {
+    $text = check_markup($text, $format)->__toString();
```

Comment:
Text from the feeds can be in markdown format, in which case we need to convert it HTML to do the sanitation. It's possible the escaping issue you experienced was due to the wrong text format id as it has changed in D9 (see comment above: "plain_text" => "markdown").